vmss-prototype/vmss-prototype

```diff
@@ -830,6 +829,25 @@ def vmss_prototype_update(sub_args):
                 '--name', extension['name']
                 ], retries=3, check=False, retry_func=not_found_no_retry)
 
+    # Scale out 1 more VMSS instance to replace the previously deleted instance
```

Comment:
This should be optional.  If autoscaling is in place, the autoscaler likely already did something like this.  Plus the unfortunate API of the VMSS does not have a "+1" operation but a "set new capacity" operation.  This leaves us with the problem that there is a race between reading the capacity and setting it that could be impacted by, say, the autoscaler, which may have just changed the capacity.

However, I do like the concept of adding a VM back for the one that we deleted but I also hope that we could remove the constraint of the one we deleted.